src/operator/timeout.ts

```diff
@@ -20,7 +20,8 @@ export function timeout<T>(this: Observable<T>, due: number | Date,
                            scheduler: Scheduler = async): Observable<T> {
   const absoluteTimeout = isDate(due);
   const waitFor = absoluteTimeout ? (+due - scheduler.now()) : Math.abs(<number>due);
-  const error = errorToSend || new TimeoutError();
+  const error = errorToSend instanceof Error ? errorToSend : new TimeoutError(errorToSend);
```

Comment:
I don't fully understand the use case for providing an error **instance**. Why would you want to do `.timeout(10, new SomeError())`? Instead, wouldn't you really want to pass the class itself, to be instantiated _when_ there is an error? `.timeout(10, SomeError)`? IMO that (along with the overload of providing just a string) is the more ideal API.